### PR TITLE
[Fix] Get parent folder when called on non-existent drive on windows

### PR DESCRIPTION
**Pull request by BrettCleary**

### CODE REVIEW

File: src/functions/getFirstExistingParentPath.ts [hunk 1]
```diff
@@ -8,15 +8,19 @@ import Dependencies from '@/src/types/dependencies'
  * @param dependencies - Dependencies container
  */
 async function getFirstExistingParentPath(directoryPath: string, dependencies: Dependencies): Promise<string> {
-	let parentDirectoryPath = directoryPath
+	let parentDirectoryPath = dependencies.pathNormalize(directoryPath)
 	let parentDirectoryFound = await isDirectoryExisting(parentDirectoryPath, dependencies)
 
-	while (!parentDirectoryFound) {
-		parentDirectoryPath = dependencies.pathNormalize(parentDirectoryPath + '/..')
+	for (let i = 0; i < 4096 && !parentDirectoryFound; ++i) {
```
Comment (eliobricenov) [inline src/functions/getFirstExistingParentPath.ts]: I'm curious why the loop limit is `4096`. Can we add a variable for it?

File: src/functions/getFirstExistingParentPath.ts [hunk 2]
```diff
@@ -8,15 +8,19 @@ import Dependencies from '@/src/types/dependencies'
...
+		const newParentDirectoryPath = dependencies.pathNormalize(parentDirectoryPath + '/..')
+		if (parentDirectoryPath === newParentDirectoryPath) {
+			return ''
```
Comment (eliobricenov) [inline src/functions/getFirstExistingParentPath.ts]: What does it mean if this function returns an empty string?

**Review by eliobricenov — APPROVED** [review]

File: src/index.ts [hunk 3]
```diff
@@ -140,6 +140,7 @@ function checkDiskSpace(directoryPath: string, dependencies: Dependencies = {
 	 * @param directoryPath - The file/folder path from where we want to know disk space
 	 */
 	async function checkUnix(directoryPath: string): Promise<DiskSpace> {
+		console.log('path ', dependencies.pathNormalize(directoryPath), ' sep ', dependencies.pathSep)
```
Comment (red-game-dev) [inline src/index.ts]: Console logging here, i guess only for debugging ?

**Review by red-game-dev — APPROVED** [review]

File: src/index.ts (hunk 3 above)
Comment (BrettCleary) [reply src/index.ts]: ah my bad. will remove

File: src/functions/getFirstExistingParentPath.ts (hunk 2 above)
Comment (BrettCleary) [reply src/functions/getFirstExistingParentPath.ts]: it failed basically. I left this behavior the same as is used in heroic. Personally I'd probably rather do `undefined` but both evaluate to false and you can't have an empty string as a directory so I it works

File: src/functions/getFirstExistingParentPath.ts (hunk 1 above)
Comment (BrettCleary) [reply src/functions/getFirstExistingParentPath.ts]: ah yeah. figured I'd get a question about this. I should have added a comment. 

While loops are error prone (and the reason we forked in the first place), so I just wanted to set a theoretical limit on the number of folders we'd have to traverse and convert to a for loop. I think the max number is < 4096 though. I'll update it and put it in a variable with a comment

Comment (BrettCleary) [reply src/functions/getFirstExistingParentPath.ts]: okay updated!

File: src/functions/getFirstExistingParentPath.ts (hunk 2 above)
Comment (BrettCleary) [reply src/functions/getFirstExistingParentPath.ts]: added a constant for this!

**Pull request merged**